### ISSUE TITLE
DefaultConvertible causing issues for Doubles

### ISSUE THREAD

**Issue by litso** [lyft/mapper#97]
The default implementation for `double` makes it impossible to allow automatic mapping from JSON that is represents floating point values a strings (which is common).

For example if I have:

```json
{
   "radius" : 200,
   "latitude" : "34.0433903",
   "kind" : "circular",
   "longitude" : "-118.4542772"
}
```

I would like to map latitude like so (without having to remember to specify a custom transform each time).

```swift
let latitude = map.from("latitude")
```

Because `DefaultConvertible` extends `Convertible` I cannot extend `Convertible` to support this.
Any thoughts?

**Comment by keith** [comment]
This issue reminds me of https://github.com/lyft/mapper/pull/86

The problem I see here is we don't really want to impose any behavior/overhead that would be unexpected for the "happy path". After the issue I linked above we discussed removing our default implementations of convertibles all together, or possibly just creating a subspec that didn't include them. Can you think of another solution here besides adding some special cases for doubles? I don't think we want to handle other situations like this since for many consumers (or at least for us) if we get a string when we expect to get a double, we've broken an API contract and we do want the parsing to fail.

**Comment by litso** [comment]
Thanks for looking at this @keith. You are right that accepting string values for doubles might not be the best solution for everyone. In our case we have a Rails API that is using BigDecimal which will usually serialize as a string value in JSON but that is my specific situation.

One solution would be to remove the default conformances in DefaultConvertible and instead leave extending the basic types up to whomever is using the library e.g.

```Swift
import Foundation
import Mapper

extension String: DefaultConvertible {}
extension Int: DefaultConvertible {}
extension Double: MyDefaultConverter{
      public static func fromMap(_ value: Any) throws -> Double {
        ...
      }
}
```

**Comment by keith** [comment]
I'm thinking about providing a subspec that allows you to opt out of any convertible implementations defined by Mapper. That way, as needed, you could define these yourself.

The only downside here is this only solves this problem for people using Mapper with CocoaPods.

**Comment by litso** [comment]
@keith awesome! thank you

**Comment by keith** [comment]
This has been released to CocoaPods as 9.0.1, please let me know if you see any issues
